cmd/operator/options/options.go

```diff
@@ -109,5 +115,30 @@ func New() *Options {
 		}
 	}
 
+	// check and set allowedServiceResources
+	if *allowedServiceResources == "" {
+		opts.AllowedServiceResources = []client.Object{}
+		return &opts
+	}
+	resourceStrs := strings.Split(*allowedServiceResources, ",")
+	resourceSet := make(map[client.Object]struct{}, len(resourceStrs))
+	for _, resource := range resourceStrs {
+		switch resource {
+		case "CronJob":
+			resourceSet[&batchv1.CronJob{}] = struct{}{}
+		case "Job":
+			resourceSet[&batchv1.Job{}] = struct{}{}
+		case "DaemonSet":
+			resourceSet[&appsv1.DaemonSet{}] = struct{}{}
+		default:
+			log.Errorf("unSupported k8s resource [%v] for service creation", resource)
+		}
+	}
+	resources := make([]client.Object, 0, len(resourceSet))
+	for resource := range resourceSet {
+		resources = append(resources, resource)
```

Comment:
Why not append `opts.AllowedServiceResources` directly?

Reply:
following is the reason for my logic
1. remove duplicate user-inputs
2. error log for unSupported k8s resource